Library/src/androidTest/java/dev/testify/AssertExpectedDeviceTest.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2023 ndtp
+  *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package dev.testify
+
+import dev.testify.annotation.ScreenshotInstrumentation
+import dev.testify.internal.exception.ScreenshotBaselineNotDefinedException
+import dev.testify.internal.exception.UnexpectedDeviceException
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+
+class AssertExpectedDeviceTest {
+
+    @get:Rule
+    val rule: ScreenshotRule<TestActivity> = ScreenshotRule(TestActivity::class.java)
+
+    @get:Rule
+    val expectedException: ExpectedException = ExpectedException.none()
+
+    /**
+     * Baseline matches current device
+     */
+    @ScreenshotInstrumentation
+    @Test
+    fun testExpectedDevice() {
+        rule.assertSame()
+    }
+
+    /**
+     * Ensure that missing a baseline is higher priority
+     */
+    @ScreenshotInstrumentation
+    @Test
+    fun testMissingBaseline() {
+        expectedException.expect(ScreenshotBaselineNotDefinedException::class.java)
```

Comment:
These tests are kind of unobvious in how they work from the code; can we add a comment or something to specify which ones are expected to be run with a different-from-project-default emulator?

Reply:
Excellent point and great feedback. I'll definitely add some explanation in some comments